### PR TITLE
Product Creation with AI: Analytics for product creation with AI

### LINKED ISSUE #9843: Tracking
pe5sF9-22S-p2#metrics

### PR DESCRIPTION
**Pull request by hafizrahman**
Closes: #9843 

Please don't merge until target branch is `trunk`

### Description

Adds the following event trackings:

| Event Name | Trigger | Properties |
|------------|---------|------------|
| *_product_creation_ai_entry_point_displayed | When the action sheet with the option to creation product with AI is displayed. |  |
| *_product_creation_ai_entry_point_tapped | When the option to create product with AI is tapped. |  |
| *_product_creation_ai_product_name_continue_button_tapped | When the Continue CTA on the product name screen is tapped. |  |
| *_product_creation_ai_tone_selected | When a tone is selected for product creation AI. | value: casual \| flowery \| convincing \| formal |
| *_product_creation_ai_generate_details_tapped | When the Generate product details is tapped on the About your product screen. | is_first_attempt: true \| false |
| *_product_creation_ai_generate_product_details_success | When generating product details with AI succeeds. |  |
| *_product_creation_ai_generate_product_details_failed | When generating product details with AI fails. | Error detail properties |
| *_product_creation_ai_save_as_draft_button_tapped | When the Save as draft button on the Preview screen is tapped. |  |
| *_product_creation_ai_save_as_draft_success | When saving an AI-generated product succeeds. |  |
| *_product_creation_ai_save_as_draft_failed | When saving an AI-generated product fails. | Error detail properties |

Extras:
- it also adds `is_ai_content` property (true/false) to the `product_detail_update_button_tapped` event, to mark that content published from this flow has AI content in it.
- also adds `ai_identify_language_success` and `ai_idenfity_language_failed` with `source` value `product_creation`.

### Testing instructions

0. Check the events in logcat:
1. Tap "Add your first product" from onboarding list, check for `product_creation_ai_entry_point_displayed`
2. Tap the product with AI option, check for `product_creation_ai_entry_point_tapped`
3. Enter a name on the next screen, tap Continue, check for `product_creation_ai_product_name_continue_button_tapped`
4. Select a tone on the next screen, check for `product_creation_ai_tone_selected` with the right value
5. Tap "Create product details" on the screen, check for `product_creation_ai_generate_details_tapped` with `is_first_attempt` set to `true`
6. Wait until generation is done, check for `product_creation_ai_generate_product_details_success`
7. Go back, then tap "Create product details" again. check for `product_creation_ai_generate_details_tapped` with `is_first_attempt` set to `false`
8. Tap "Save as draft" at the top, check for `product_creation_ai_save_as_draft_success` once product details screen is loaded.

**Error cases:**
Try re-doing step 6 and step 8 but trigger error (turn off internet for 6).

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AddProductWithAIViewModel.kt [hunk 1]
```diff
@@ -90,27 +89,35 @@ class AddProductWithAIViewModel @Inject constructor(
...
         require(::product.isInitialized)
         viewModelScope.launch {
             saveButtonState.value = SaveButtonState.Loading
             val (success, productId) = productDetailRepository.addProduct(product)
             if (!success) {
                 triggerEvent(ShowSnackbar(R.string.error_generic))
                 saveButtonState.value = SaveButtonState.Shown
+                AnalyticsTracker.track(
+                    AnalyticsEvent.PRODUCT_CREATION_AI_SAVE_AS_DRAFT_FAILED,
+                    mapOf(AnalyticsTracker.KEY_ERROR_CONTEXT to this::class.java.simpleName)
```
Comment (hafizrahman) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AddProductWithAIViewModel.kt]: Not sure what other error details we can add here, if `productDetailRepository.addProduct()` doesn't give any.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/ProductPreviewSubViewModel.kt [hunk 2]
```diff
@@ -114,13 +114,22 @@ class ProductPreviewSubViewModel(
...
+                    AnalyticsTracker.track(AnalyticsEvent.PRODUCT_CREATION_AI_GENERATE_PRODUCT_DETAILS_SUCCESS)
                     _state.value = State.Success(
                         product = product,
                         propertyGroups = buildProductPreviewProperties(product)
                     )
                     onDone(product)
                 },
                 onFailure = {
+                    AnalyticsTracker.track(
+                        AnalyticsEvent.PRODUCT_CREATION_AI_GENERATE_PRODUCT_DETAILS_SUCCESS,
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/ProductPreviewSubViewModel.kt]:
```suggestion
                        AnalyticsEvent.PRODUCT_CREATION_AI_GENERATE_PRODUCT_DETAILS_FAILED,
```

Comment (hafizrahman) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/ProductPreviewSubViewModel.kt]: Was added on https://github.com/woocommerce/woocommerce-android/pull/9858/commits/0a696e7d00744123cb751f0414768bba76774f6f instead, good catch! I think we found it around the same time.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AddProductWithAIViewModel.kt (hunk 1 above)
Comment (JorgeMucientes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AddProductWithAIViewModel.kt]: Hmm, I guess I'd only track `AnalyticsEvent.PRODUCT_CREATION_AI_SAVE_AS_DRAFT_FAILED,` and remove the added properties. With the current code we're just adding the `CoroutineScope` class name. 

Imho we can remove this line:
```
mapOf(AnalyticsTracker.KEY_ERROR_CONTEXT to this::class.java.simpleName)
```

**Review by JorgeMucientes — APPROVED** [review]

**Comment by JorgeMucientes** [comment]
Nice job @hafizrahman 👏🏼 ! Thanks for handling this, everything works as expected now. I just a very minor np about the error property value passed for `PRODUCT_CREATION_AI_SAVE_AS_DRAFT_FAILED` event tracking. Let me know what you think.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AddProductWithAIViewModel.kt (hunk 1 above)
Comment (hafizrahman) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AddProductWithAIViewModel.kt]: resolved in https://github.com/woocommerce/woocommerce-android/pull/9858/commits/bb86616d1a8f0d413934a8a7ee1751302df3f497

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/AIProductDescriptionViewModel.kt [hunk 3]
```diff
@@ -79,7 +80,7 @@ class AIProductDescriptionViewModel @Inject constructor(
         }
 
         tracker.track(
-            stat = PRODUCT_DESCRIPTION_AI_GENERATION_SUCCESS,
+            stat = AI_IDENTIFY_LANGUAGE_SUCCESS,
```
Comment (hafizrahman) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/AIProductDescriptionViewModel.kt:83]: Unrelated to this PR, just something wrong that I noticed in a different file.

**Pull request merged**